src/js/actions/export.js

```diff
@@ -0,0 +1,365 @@
+     * @return {Promise}
+     */
+    var updateLayerExportAsset = function (document, layer, assetIndex, props) {
+        var documentID = document.id,
+            assetPropsArray = [],
+            payload;
+
+        assetPropsArray[assetIndex] = props;
+
+        payload = {
+            documentID: documentID,
+            layerIDs: layer.id,
+            assetPropsArray: assetPropsArray
+        };
+
+        return this.dispatchAsync(events.export.ASSET_CHANGED, payload).bind(this).then(function () {
+            return _syncLayerExportMetadata.call(this, documentID, layer.id);
+        });
+    };
+    updateLayerExportAsset.reads = [locks.JS_DOC];
+    updateLayerExportAsset.writes = [locks.JS_DOC, locks.PS_DOC];
+
+    /**
+     * Set the numerical scale of the asset specified by the given index
+     *
+     * @param {Document} document
+     * @param {Layer} layer
+     * @param {number} assetIndex index of this asset within the layer's list
+     * @param {number} scale
+     * @return {Promise}
+     */
+    var updateLayerAssetScale = function (document, layer, assetIndex, scale) {
+        return this.transfer(updateLayerExportAsset, document, layer, assetIndex, { scale: scale || 1 });
+    };
+    updateLayerAssetScale.reads = [];
+    updateLayerAssetScale.writes = [];
+    updateLayerAssetScale.transfers = [updateLayerExportAsset];
+
+    /**
+     * Set the filename suffix of the asset specified by the given index
+     *
+     * @param {Document} document
+     * @param {Layer} layer
+     * @param {number} assetIndex index of this asset within the layer's list
+     * @param {string} suffix
+     * @return {Promise}
+     */
+    var updateLayerAssetSuffix = function (document, layer, assetIndex, suffix) {
+        return this.transfer(updateLayerExportAsset, document, layer, assetIndex, { suffix: suffix });
+    };
+    updateLayerAssetSuffix.reads = [];
+    updateLayerAssetSuffix.writes = [];
+    updateLayerAssetSuffix.transfers = [updateLayerExportAsset];
+
+    /**
+     * Set the format of the asset specified by the given index
+     *
+     * @param {Document} document
+     * @param {Layer} layer
+     * @param {number} assetIndex index of this asset within the layer's list
+     * @param {string} format (example: jpg)
+     * @return {Promise}
+     */
+    var updateLayerAssetFormat = function (document, layer, assetIndex, format) {
+        return this.transfer(updateLayerExportAsset, document, layer, assetIndex, { format: format });
+    };
+    updateLayerAssetFormat.reads = [];
+    updateLayerAssetFormat.writes = [];
+    updateLayerAssetFormat.transfers = [updateLayerExportAsset];
+
+    /**
+     * Update the status of all assets that are being requested
+     *
+     * @param {Document} document [description]
+     */
+    var setAllAssetsRequested = function (document) {
+        var layerIDs = document.layers.all.reduce(function (IDs, layer) {
```

Comment:
If it suits your fancy, this could be more structured with `.toSeq`/`.filter`/`.map`/`.toJS`